File: src/main/java/in/tazj/k8s/letsencrypt/model/CertificateResponse.java
package in.tazj.k8s.letsencrypt.model;

import java.util.Date;
import java.util.List;
import java.util.Map;

import lombok.Value;

/**
 * A signed certificate as returned by the ACME server.
 */
@Value
public class CertificateResponse {
  List<String> domains;
  Map<String, String> certificateFiles;
  Date expiryDate;
  String ca;
}

File: src/main/kotlin/in/tazj/k8s/letsencrypt/model/CertificateResponse.kt
package `in`.tazj.k8s.letsencrypt.model

import java.util.*

/**
 * A signed certificate as returned by the ACME server.
 */
data class CertificateResponse(
        val domains: List<String>,
        val certificateFiles: Map<String, String>,
        val expiryDate: Date,
        val ca: String
)